### ISSUE TITLE
Typescript: rest operator is removed in array destructuring

### ISSUE THREAD

**Issue by steadicat** [prettier/prettier#1870]
```diff
- function test([first, ...rest]) {}
+ function test([first, rest]) {}
```
https://prettier.github.io/prettier/#%7B%22content%22%3A%22function%20test(%5Bfirst%2C%20...rest%5D)%20%7B%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22typescript%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

**Comment by vjeux** [comment]
Oh noes, we need to fix it.

You can add `// prettier-ignore` on the line before while we get a fix out. Sorry about that! (Also glad that you are looking into using it!!)

**Comment by steadicat** [comment]
Oh we are definitely using it! Already ran it on the whole codebase and added it to our Makefile. Just that one spot needed a `// prettier-ignore`. Thanks for making this work with TypeScript! So exciting. :)

**Comment by vjeux** [comment]
Just reported the issue upstream: https://github.com/eslint/typescript-eslint-parser/issues/306

**Comment by vjeux** [comment]
I just published 1.4.1 which fixes it!
